File: src/main/java/com/ajax/springcourse/car/model/dto/CarCreateDto.java
package com.ajax.springcourse.car.model.dto;

import com.ajax.springcourse.car.model.Car;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import javax.validation.constraints.Min;
import javax.validation.constraints.NotNull;

@Getter
@Setter
@NoArgsConstructor
public class CarCreateDto {

    @NotNull
    private String brand;
    @NotNull
    private String model;
    @NotNull
    @Min(1)
    private int seats;
    @NotNull
    private String description;

    public CarCreateDto(Car car) {
        brand = car.getBrand();
        model = car.getModel();
        seats = car.getSeats();
        description = car.getDescription();
    }

    public Car mapToCar() {
        return new Car(brand, model, seats, description);
    }
}

File: src/main/java/com/ajax/springcourse/car/model/dto/CarCreateDto.kt
package com.ajax.springcourse.car.model.dto

import com.ajax.springcourse.car.model.Car
import javax.validation.constraints.Min

class CarCreateDto(
    var brand: String = "",
    var model: String = "",
    var seats: @Min(1) Int = 1,
    var description: String = ""
) {
    fun mapToCar(): Car = Car(
        brand = brand,
        model = model,
        seats = seats,
        description = description
    )
}